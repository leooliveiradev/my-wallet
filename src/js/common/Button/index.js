import styled from 'styled-components';
import { fromTheme } from 'app/js/utils';

const Button = styled.button.attrs({
  type: ({ type }) => type,
})`
  display: inline-block;
  padding: 0 1.4rem;
  height: 2.8rem;
  border-radius: 4rem;
  border: none;
  outline: none;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
  background-color: ${fromTheme('color.primary')};
  font-family: ${fromTheme('fontStyle.primary')};
  font-size: 1.2rem;
  color: #fff;
`;

Button.defaultProps = {
  type: 'button',
};

export default Button;
