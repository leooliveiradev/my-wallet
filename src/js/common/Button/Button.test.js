import React from 'react';
import Button from 'app/js/common/Button';
import renderer from 'react-test-renderer';

describe('Button snapshot', () => {
  it('should render the Button', () => {
    const component = renderer.create(
      <Button>
        Add transaction
      </Button>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
