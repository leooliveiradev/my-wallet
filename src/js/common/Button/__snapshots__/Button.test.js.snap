// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot should render the Button 1`] = `
<button
  className="sc-bdVaJa jolRSw"
  type="button"
>
  Add transaction
</button>
`;
